### ISSUE TITLE
how could i get the 'es' value in my dataset?

### ISSUE THREAD

**Issue by TqXue** [MacLLL/SELC#2]
Really good jobs!
But, there are some questions as follows:
(1) if i use your method in my dataset, how could i calculate the value of 'es'? Furthermore, how to plot the loss distribution in Figure2?
(2) whether the `pred_detach` in `pred_detach = F.softmax(logits.detach(), dim=1)` in real-world_noise/train_Clothing1M.py line60 stands for the `ensemble prediction` , i can't undstand why it represents `ensemble prediction`, it is very kind of you to help me.
Thank you very much for your early reply!

**Comment by MacLLL** [comment]
Thank you for interesting in our work.
(1) The es is estimated by the paper section 4.4, you need to train the model once and monitor the proposed metrics. Es is corresponding to the epoch that metric reaches to maximum value, for example, metric 3 in Equation 7.
(2)  pred_detach is the prediction in current epoch (corresponding to the p in paper). The ensemble prediction is actually the self.soft_labels (corresponding t in the paper).
(3) As for how to plot loss distribution. It is simple, you save all the per-sample loss value during training and then plot it in whatever epoch you want, just for analysis.

**Comment by TqXue** [comment]
> Thank you for interesting in our work. (1) The es is estimated by the paper section 4.4, you need to train the model once and monitor the proposed metrics. Es is corresponding to the epoch that metric reaches to maximum value, for example, metric 3 in Equation 7. (2) pred_detach is the prediction in current epoch (corresponding to the p in paper). The ensemble prediction is actually the self.soft_labels (corresponding t in the paper). (3) As for how to plot loss distribution. It is simple, you save all the per-sample loss value during training and then plot it in whatever epoch you want, just for analysis.

Thank you for your prompt reply！Since my code ability is limited, could you please show me the code for evaluating `es` and plotting the loss distribution？Sorry to bother you again. Your help is a great encouragement to me.
